src/common/task.c

```diff
@@ -72,15 +106,35 @@ task_spec *alloc_task_spec(function_id function_id,
   return task;
 }
 
+void add_task_and_return_ids(task_spec *spec) {
```

Comment:
Is it possible to always call this function in `alloc_task_spec`? That way we don't have to make sure we always call this function after creating a task spec.

Also, it would be nice to make this function idempotent (which I don't think it is right now?). If everything in the spec besides the task ID and return IDs are the same, then it should be okay to call this function more than once.